tests/dragonfly/cluster_test.py

```diff
@@ -2194,3 +2202,67 @@ async def test_cluster_memory_consumption_migration(df_factory: DflyInstanceFact
     await push_config(json.dumps(generate_config(nodes)), [node.admin_client for node in nodes])
 
     await check_for_no_state_status([node.admin_client for node in nodes])
+
+
+@pytest.mark.asyncio
+@dfly_args({"proactor_threads": 4, "cluster_mode": "yes"})
+async def test_migration_timeout_on_sync(df_factory: DflyInstanceFactory, df_seeder_factory):
+    # setting migration_timeout to a very small value to force the replica to timeout
+    instances = [
+        df_factory.create(
+            port=BASE_PORT + i,
+            admin_port=BASE_PORT + i + 1000,
+            migration_timeout=100,
+            vmodule="outgoing_slot_migration=9,cluster_family=9,incoming_slot_migration=9",
+        )
+        for i in range(2)
+    ]
+
+    df_factory.start_all(instances)
+
+    nodes = [(await create_node_info(instance)) for instance in instances]
+    nodes[0].slots = [(0, 16383)]
+    nodes[1].slots = []
+
+    await push_config(json.dumps(generate_config(nodes)), [node.admin_client for node in nodes])
+
+    logging.debug("source node DEBUG POPULATE")
+    await nodes[0].client.execute_command("debug", "populate", "200000", "foo", "1000")
+
+    seeder = df_seeder_factory.create(port=nodes[0].instance.port, cluster_mode=True)
+    capture = await seeder.capture()
```

Comment:
where is seeder run?

Reply:
If I run a seeder I can not do capture, because we don't know exact moment when the migration is finished

Reply:
also seeder capture should be after we finish the migration and you stop the seeder